packages/serverpod/lib/src/server/websocket_request_handlers/helpers/method_stream_manager.dart

```diff
@@ -0,0 +1,560 @@
+          const StreamClosedWithErrorException(),
+        );
+      }
+
+      return _closeControllers([context.controller]);
+    }
+  }
+
+  void createStream({
+    required MethodStreamCallContext methodStreamCallContext,
+    required UuidValue methodStreamId,
+    required Session session,
+  }) async {
+    var outputStreamContext = await _createOutputController(
+      methodStreamCallContext,
+      session,
+      methodStreamId,
+    );
+
+    var inputStreams =
+        _createInputStreams(methodStreamCallContext, methodStreamId);
+    var streamParams = inputStreams.map(
+      (key, value) => MapEntry(key, value.stream),
+    );
+
+    switch (methodStreamCallContext.method.returnType) {
+      case MethodStreamReturnType.streamType:
+        _handleMethodWithStreamReturn(
+          methodStreamCallContext: methodStreamCallContext,
+          session: session,
+          streamParams: streamParams,
+          outputController: outputStreamContext.controller,
+          subscription: outputStreamContext.subscription,
+          methodStreamId: methodStreamId,
+        );
+        break;
+      case MethodStreamReturnType.futureType:
+      case MethodStreamReturnType.voidType:
+        await _handleMethodWithFutureReturn(
+          methodStreamCallContext: methodStreamCallContext,
+          session: session,
+          streamParams: streamParams,
+          outputController: outputStreamContext.controller,
+          methodStreamId: methodStreamId,
+        );
+        break;
+    }
+  }
+
+  /// Dispatches a message to the correct stream controller.
+  bool dispatchData({
+    required String endpoint,
+    required String method,
+    required UuidValue methodStreamId,
+    String? parameter,
+    required Object? value,
+  }) {
+    var streamContext = _inputStreamContexts[_buildStreamKey(
+      endpoint: endpoint,
+      method: method,
+      parameter: parameter,
+      methodStreamId: methodStreamId,
+    )];
+
+    if (streamContext == null) {
+      return false;
+    }
+
+    streamContext.controller.add(value);
+    return true;
+  }
+
+  /// Dispatches an error to the correct stream controller.
+  bool dispatchError({
+    required String endpoint,
+    required String method,
+    required UuidValue methodStreamId,
+    String? parameter,
+    required Object error,
+  }) {
+    var streamContext = _inputStreamContexts[_buildStreamKey(
+      endpoint: endpoint,
+      method: method,
+      parameter: parameter,
+      methodStreamId: methodStreamId,
+    )];
+
+    if (streamContext == null) {
+      return false;
+    }
+
+    streamContext.controller.addError(error);
+    return true;
+  }
+
+  Future<_OutputStreamContext> _createOutputController(
+    MethodStreamCallContext methodStreamCallContext,
+    Session session,
+    UuidValue methodStreamId,
+  ) async {
+    bool isCancelled = false;
+    var revokedAuthenticationHandler =
+        await _RevokedAuthenticationHandler.create(
```

Comment:
Discussion: The create method communicates that this object is always created. Should some other name be used to indicate that it might not be created and null could be returned? For example `tryCreate` or something like that?

Reply:
You're right, it doesn't fit perfectly right now. I think `createIfAuthenticationIsRequired` is very explicit and explains clearly what will happen.

Reply:
1b1a0763